sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIO.java

```diff
@@ -347,29 +403,65 @@ public static Read read() {
    * sample parse function that parses click events from a table.
    *
    * <pre>{@code
-   * p.apply(BigQueryIO.read(new SerializableFunction<SchemaAndRecord, ClickEvent>() {
-   *   public Event apply(SchemaAndRecord record) {
-   *     GenericRecord r = record.getRecord();
-   *     return new Event((Long) r.get("userId"), (String) r.get("url"));
-   *   }
-   * }).from("...");
-   * }</pre>
+   * p.apply(
+   *   BigQueryIO.readAvroSource(
+   *     new SerializableFunction<SchemaAndRecord, ClickEvent>() {
+   *       public Event apply(SchemaAndRecord record) {
+   *         GenericRecord r = record.getRecord();
+   *         return new Event((Long) r.get("userId"), (String) r.get("url"));
+   *       }
+   *     }).from("..."));
+   * } </pre>
+   *
    */
-  public static <T> TypedRead<T> read(
+  public static <T> TypedRead<T> readAvroSource(
       SerializableFunction<SchemaAndRecord, T> parseFn) {
     return new AutoValue_BigQueryIO_TypedRead.Builder<T>()
         .setValidate(true)
         .setWithTemplateCompatibility(false)
         .setBigQueryServices(new BigQueryServicesImpl())
-        .setParseFn(parseFn)
+        .setAvroParseFn(parseFn)
+        .setMethod(TypedRead.Method.GCS_EXPORT)
+        .build();
+  }
+
+  /**
+   * Reads from a BigQuery table using the BigQuery parallel read API and returns a
+   * {@link PCollection} with one element per each row of the table, parsed from the BigQuery
+   * {@link com.google.cloud.bigquery.v3.RowOuterClass.Row} format using the specified function.
+   *
+   * <p>Each {@link SchemaAndRowProto} contains a BigQuery
+   * {@link com.google.cloud.bigquery.v3.RowOuterClass.StructType} representing the table schema
+   * and a {@link com.google.cloud.bigquery.v3.RowOuterClass.Row} representing the row, indexed by
+   * column name. Here is a sample parse function that parses click events from a table.
+   *
+   * <pre>{@code
+   * p.apply(
+   *   BigQueryIO.readRowProtoSource(
+   *     new SerializableFunction<SchemaAndRowProto, ClickEvent>() {
```

Comment:
Use a lambda here, and maybe in the `readViaAvro` example too?

Reply:
Done.